snapshot.go

```diff
@@ -249,3 +252,35 @@ func (r *Raft) compactLogs(snapIdx uint64) error {
 	}
 	return nil
 }
+
+// resetMonotonicLogs truncates all logs for MonotonicLogStores after a snapshot
+// restore.
+func (r *Raft) resetMonotonicLogs() error {
+	// Only truncate if the underlying store has monotonic log index
+	// requirements.
+	logs, ok := r.logs.(MonotonicLogStore)
+	if ok && logs.IsMonotonic() {
+		defer metrics.MeasureSince([]string{"raft", "resetMonotonicLogs"}, time.Now())
+	} else {
+		r.logger.Info("skipping truncate for non-monotonic log store")
+		return errNonMonotonicLogStore
+	}
+
+	// Determine log ranges to truncate
+	minLogIdx, err := r.logs.FirstIndex()
+	if err != nil {
+		return fmt.Errorf("failed to get first log index: %w", err)
+	}
+
+	lastLogIdx, err := r.logs.LastIndex()
+	if err != nil {
+		return fmt.Errorf("failed to get last log index: %w", err)
+	}
+
+	r.logger.Info("truncating monotonic log store", "from", minLogIdx, "to", lastLogIdx)
```

Comment:
```suggestion
	r.logger.Info("resetting log store", "from", minLogIdx, "to", lastLogIdx)
```
Just to make it more meaningful for operators who don't know about this one weird abstraction detail...